### PR TITLE
Concatenated input support

### PR DESCRIPTION
**Pull request by xerial**
#103

### CODE REVIEW

File: src/main/java/org/xerial/snappy/SnappyInputStream.java [hunk 1]
```diff
@@ -93,20 +92,26 @@ protected void readHeader() throws IOException {
             return;
         }
 
+        if(!isValidHeader(header)) {
```
Comment (JoshRosen) [inline src/main/java/org/xerial/snappy/SnappyInputStream.java]: This branch is never taken in unit tests, suggesting a test coverage issue.

File: src/main/java/org/xerial/snappy/SnappyInputStream.java [hunk 2]
```diff
@@ -331,16 +356,24 @@ protected boolean hasNextChunk() throws IOException {
...
+        int readBytes = readNext(header, 0, 4);
+        if(readBytes < 4)
+            return false;
+
+        int chunkSize = SnappyOutputStream.readInt(header, 0);
+        if(chunkSize == SnappyCodec.MAGIC_HEADER_HEAD) {
+            // Concatenated data
+            int remainingHeaderSize = SnappyCodec.headerSize() - 4;
+            readBytes = readNext(header, 4, remainingHeaderSize);
+            if(readBytes < remainingHeaderSize)
```
Comment (JoshRosen) [inline src/main/java/org/xerial/snappy/SnappyInputStream.java]: This branch is never exercised by unit tests.  Also, it should probably have curly braces to guard against bugs if more things are ever added to this branch.

File: src/main/java/org/xerial/snappy/SnappyCodec.java [hunk 3]
```diff
@@ -52,6 +52,13 @@
     public static final byte[] MAGIC_HEADER               = new byte[] { -126, 'S', 'N', 'A', 'P', 'P', 'Y', 0 };
     public static final int    MAGIC_LEN                  = MAGIC_HEADER.length;
     public static final int    HEADER_SIZE                = MAGIC_LEN + 8;
+    public static final int    MAGIC_HEADER_HEAD         = SnappyOutputStream.readInt(MAGIC_HEADER, 0);
+    public static final int    MAGIC_HEADER_TAIL         = SnappyOutputStream.readInt(MAGIC_HEADER, 4);
```
Comment (JoshRosen) [inline src/main/java/org/xerial/snappy/SnappyCodec.java]: This variable is unused.

File: src/main/java/org/xerial/snappy/SnappyInputStream.java (hunk 2 above)
Comment (xerial) [inline src/main/java/org/xerial/snappy/SnappyInputStream.java]: good point. We should add coverage report (jacoco, etc.) and some coding style enforcer to use curly braces for single statement 'if' blocks.

File: src/main/java/org/xerial/snappy/SnappyCodec.java (hunk 3 above)
Comment (xerial) [inline src/main/java/org/xerial/snappy/SnappyCodec.java]: will remove it.

File: src/main/java/org/xerial/snappy/SnappyInputStream.java (hunk 1 above)
Comment (xerial) [inline src/main/java/org/xerial/snappy/SnappyInputStream.java]: oh really? I'll add a test to read the result of Snappy.compress with SnappyInputStream.

**Pull request merged**